components/LocationInput.tsx

```diff
@@ -80,18 +89,30 @@ const LocationInput: FC<PropsWithChildren<LocationInputProps>> = ({
         setLat(newLat);
         setLngExternal(newLng);
         setLatExternal(newLat);
+
+        savedLat = newLat;
+        savedLng = newLng;
+
         setZoom(map.current!.getZoom());
         updateMarker(marker, map.current!, zoomLevelThreshold, newLng, newLat);
       };
 
       map.current.on('move', moveHandler);
 
-      map.current.on('moveend', () =>
-        setIsSubmissionEnabled(map.current!.getZoom() > zoomLevelThreshold)
-      );
+      map.current.on('moveend', () => {
+        const currentZoom = map.current!.getZoom();
+
+        if (currentZoom >= reverseGeocodingZoomThreshold) {
+          geocoder.query(`${savedLat}, ${savedLng}`);
+        }
+
+        return setIsSubmissionEnabled(currentZoom > zoomLevelThreshold);
+      });
     }
   });
```

Comment:
I'm now realizing this `useEffect` doesn't declare a dependency array of `[]`... this is probably not what we want. I see this was here before, and everything does work. Maybe in the future we fix this. I could make an issue or you could try with it.